### ISSUE TITLE
minikube not starting with Docker Desktop 4.16.0

### ISSUE THREAD

**Issue by spowelljr** [kubernetes/minikube#15634]
Trying to start minikube with Docker Desktop 4.16.0 results in the following
```
$ ./out/minikube start --driver docker
😄  minikube v1.28.0 on Darwin 13.1 (arm64)
✨  Using the docker driver based on user configuration
📌  Using Docker Desktop driver with root privileges
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
✋  Stopping node "minikube"  ...
🛑  Powering off "minikube" via SSH ...
🔥  Deleting "minikube" in docker ...
🤦  StartHost failed, but will try again: creating host: create: provisioning: ssh command error:
command : sudo hostname minikube && echo "minikube" | sudo tee /etc/hostname
err     : Process exited with status 1
output  : sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?

🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
😿  Failed to start docker container. Running "minikube delete" may fix it: creating host: create: provisioning: ssh command error:
command : sudo hostname minikube && echo "minikube" | sudo tee /etc/hostname
err     : Process exited with status 1
output  : sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?


❌  Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: provisioning: ssh command error:
command : sudo hostname minikube && echo "minikube" | sudo tee /etc/hostname
err     : Process exited with status 1
output  : sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?


╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯


```

**Comment by medyagh** [comment]
We are currently working to figure out the root cause for this issue. meanwhile if you have this issue please revert Docker Desktop to v4.15.0 as the last working verison.

**Comment by medyagh** [comment]
this seems to be related to https://github.com/docker/for-mac/issues/6675

**Comment by medyagh** [comment]
@prezha found out 
```
looks like it needs [un]setting no_new_privs:
dd v4.16.0:
➜  ~ minikube ssh -- setpriv -d
uid: 1000
euid: 1000
gid: 999
egid: 999
Supplementary groups: 27,109,111,999
no_new_privs: 1
Inheritable capabilities: [none]
Ambient capabilities: [none]
Capability bounding set: chown,dac_override,dac_read_search,fowner,fsetid,kill,setgid,setuid,setpcap,linux_immutable,net_bind_service,net_broadcast,net_admin,net_raw,ipc_lock,ipc_owner,sys_module,sys_rawio,sys_chroot,sys_ptrace,sys_pacct,sys_admin,sys_boot,sys_nice,sys_resource,sys_time,sys_tty_config,mknod,lease,audit_write,audit_control,setfcap,mac_override,mac_admin,syslog,wake_alarm,block_suspend,audit_read
Securebits: [none]
Parent death signal: [none]
dd v4.15.0:
➜  ~ minikube ssh -- setpriv -d
uid: 1000
euid: 1000
gid: 999
egid: 999
Supplementary groups: 27,109,111,999
no_new_privs: 0
Inheritable capabilities: [none]
Ambient capabilities: [none]
Capability bounding set: chown,dac_override,dac_read_search,fowner,fsetid,kill,setgid,setuid,setpcap,linux_immutable,net_bind_service,net_broadcast,net_admin,net_raw,ipc_lock,ipc_owner,sys_module,sys_rawio,sys_chroot,sys_ptrace,sys_pacct,sys_admin,sys_boot,sys_nice,sys_resource,sys_time,sys_tty_config,mknod,lease,audit_write,audit_control,setfcap,mac_override,mac_admin,syslog,wake_alarm,block_suspend,audit_read
Securebits: [none]
Parent death signal: [none]
ref: https://man7.org/linux/man-pages/man1/setpriv.1.html (
```

it's mentioned [here](https://docs.docker.com/engine/security/non-events/):

**Comment by spowelljr** [comment]
Updating to 4.16.1 will resolve this issue

https://docs.docker.com/desktop/release-notes/#4161

**Comment by spowelljr** [comment]
Since this is solved and users updating now will get 4.16.2 I'm going to close this issue.
